### PR TITLE
genesis: begin reworking genesis to enable deterministic initialization of initial chain state

### PR DESCRIPTION
**Pull request by bmwill**
Today, in order to initialize a validator's initial chain state we need to disseminate a "GenesisConfig" which the validator then attempts to re-perform the entire genesis process from scratch. Unfortunately this process can be non-deterministic if configured to generate accounts without a pre-set address.

This PR changes this process by instead allowing for the genesis ceremony to be performed a single time and then have the resulting state packaged in a `Genesis` struct, which is then included in a validator's `ValidatorConfig`. This resulting state can be used to deterministically populate the initial state of the chain. More specifically, when a validator starts up and its storage is empty, it will process the contents of the `Genesis` struct to populate storage with the initial state of the chain.

This builds off of https://github.com/MystenLabs/sui/pull/1939 and implements the missing piece required for deployment as a validator's `ValidatorConfig` will have a complete picture of everything a validator will need when starting up.

There is still a bit more work to be done to make the genesis process a bit cleaner. Ideally genesis would be a single Transaction Certificate and this is a step toward that reality.

### CODE REVIEW

File: sui/src/unit_tests/cli_tests.rs [hunk 1]
```diff
@@ -403,7 +403,7 @@ async fn test_custom_genesis_with_custom_move_package() -> Result<(), anyhow::Er
     // Checks network config contains package ids
     let network_conf =
         PersistedConfig::<NetworkConfig>::read(&working_dir.join(SUI_NETWORK_CONFIG))?;
-    assert_eq!(2, network_conf.loaded_move_packages().len());
+    // assert_eq!(2, network_conf.loaded_move_packages().len());
```
Comment (oxade) [inline sui/src/unit_tests/cli_tests.rs:406]: Remove?

File: crates/sui-config/src/genesis.rs [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+        // Load Move Framework
+        let move_framework_lib_path = self.move_framework.unwrap();
+        info!(
+            "Loading Move framework lib from {:?}",
+            move_framework_lib_path
+        );
+        let move_modules =
+            sui_framework::get_move_stdlib_modules(&move_framework_lib_path).unwrap();
+        // let move_framework =
+        //     Object::new_package(move_modules.clone(), TransactionDigest::genesis());
```
Comment (oxade) [inline crates/sui-config/src/genesis.rs:146]:
```suggestion
```

File: crates/sui-config/src/genesis.rs [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+
+    pub fn genesis_ctx(mut self, genesis_ctx: TxContext) -> Self {
+        self.genesis_ctx = Some(genesis_ctx);
+        self
+    }
+
+    // pub fn add_account(mut self, config: AccountConfig) -> Self {
+    //     self.accounts.push(config);
+    //     self
+    // }
```
Comment (oxade) [inline crates/sui-config/src/genesis.rs:125]:
```suggestion
```

File: crates/sui-config/src/genesis.rs [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+        info!(
+            "Loading Move framework lib from {:?}",
+            move_framework_lib_path
+        );
+        let move_modules =
+            sui_framework::get_move_stdlib_modules(&move_framework_lib_path).unwrap();
+        // let move_framework =
+        //     Object::new_package(move_modules.clone(), TransactionDigest::genesis());
+        modules.push(move_modules);
+        // objects.push(move_framework);
```
Comment (oxade) [inline crates/sui-config/src/genesis.rs:148]:
```suggestion
```

File: crates/sui-config/src/genesis.rs [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+
+        // Load Sui Framework
+        let sui_framework_lib_path = self.sui_framework.unwrap();
+        info!(
+            "Loading Sui framework lib from {:?}",
+            sui_framework_lib_path
+        );
+        let sui_modules =
+            sui_framework::get_sui_framework_modules(&sui_framework_lib_path).unwrap();
+        // let sui_framework = Object::new_package(sui_modules.clone(), TransactionDigest::genesis());
```
Comment (oxade) [inline crates/sui-config/src/genesis.rs:158]:
```suggestion
```

File: crates/sui-config/src/genesis.rs [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+        modules.push(sui_modules);
+        // objects.push(sui_framework);
```
Comment (oxade) [inline crates/sui-config/src/genesis.rs:160]:
```suggestion
```

**Comment by bmwill** [comment]
most of the things commented out were mostly around a "we probably want to do this but so I don't forget i'll comment it out". Which should be fixed on a follow up.

**Pull request merged**